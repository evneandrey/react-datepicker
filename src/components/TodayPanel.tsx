import * as React from 'react';
import * as classNames from 'classnames';

interface IProps {
  /** panel display today string */
  today: string;
  /** today panel click event */
  onClick?: () => void;
  /** today panel show or hide */
  show?: boolean;
}

const TodayPanel: React.FunctionComponent<IProps> = ({ today, show, onClick }) => (
  <div className={classNames('calendar__panel--today', { 'calendar__panel--show': show })}>
    <h2 onClick={onClick}>{today}</h2>
  </div>
);

export default TodayPanel;
