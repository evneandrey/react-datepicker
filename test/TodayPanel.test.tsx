import * as React from 'react';
import * as sinon from 'sinon';
import TodayPanel from '../src/components/TodayPanel';
import { shallow, ShallowWrapper } from 'enzyme';

describe('<TodayPanel/>', () => {
  let component: ShallowWrapper<React.Component>;
  let onClick: sinon.SinonSpy;

  beforeEach(() => {
    onClick = sinon.spy();
    component = shallow(<TodayPanel onClick={onClick} today="20190101" show />);
  });

  it('renders without props', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('props show correctly', () => {
    expect(component.hasClass('calendar__panel--show')).toBeTruthy();
  });

  it('props onClick correctly', () => {
    component.find('h2').simulate('click');
    expect(onClick).toHaveProperty('callCount', 1);
  });

  it('props today correctly', () => {
    expect(component.find('h2').text()).toEqual('20190101');
  });
});
