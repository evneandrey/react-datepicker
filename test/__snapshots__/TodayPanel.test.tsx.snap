// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodayPanel/> renders without props 1`] = `
<div
  className="calendar__panel--today calendar__panel--show"
>
  <h2
    onClick={[Function]}
  >
    20190101
  </h2>
</div>
`;
